include/psa/crypto_values.h

```diff
@@ -1255,6 +1278,79 @@
 #define PSA_ALG_IS_RANDOMIZED_ECDSA(alg)                                \
     (PSA_ALG_IS_ECDSA(alg) && !PSA_ALG_ECDSA_IS_DETERMINISTIC(alg))
 
+/** Edwards-curve digital signature algorithm without prehashing (PureEdDSA),
+ * using standard parameters.
+ *
+ * Contexts are not supported in the current version of this specification
+ * because there is no suitable signature interface that can take the
+ * context as a parameter. A future version of this specification may add
+ * suitable functions and extend this algorithm to support contexts.
+ *
+ * PureEdDSA requires an elliptic curve key on a twisted Edwards curve.
+ * In this specification, the following curves are supported:
+ * - #PSA_ECC_FAMILY_TWISTED_EDWARDS, 255-bit: Ed25519 as specified
```

Comment:
The key type says 256-bit. I'm not sure which one is right.

Reply:
Hmm, weird question. One could argue that the key is really only 251 bits. People seem to call it 256-bits though, even though, during key generation, 5 bits are always set to specific values.

Reply:
One could argue the curve size is 255, since operations on the curve use modulo p which is 2^255 - 19. But it's a discussion about semantics.

Reply:
I don't think we need to be pedantic here. For some users reading 255 or 251 might be alarming, raising questions. I think that this time we could be nice to users and say 256. This means 2-3 bit differences in implied bit security, but that shouldn't matter in the overwhelming majority of use cases. (I know, this is a complete opposite what I was saying 2 years ago.)

Reply:
The existing bit size in PSA for Curve25519 is 255: https://github.com/ARMmbed/mbedtls/blob/7f3d10de02567954077095f28211d8f0aa53c96f/library/psa_crypto.c#L457

Since Curve25519 and Ed25519 are equivalent, can we be consistent and say 255 for Ed25519 as well?

Reply:
The generic definition in the PSA specification is “the bit size of ``q`` for a curve over ``F_q``” for Weierstrass curves. There's a different definition for Montgomery curves, but it gives the same results for Curve25519 and Ed25519. I don't see a reason to use different sizes for Curve25519 and Ed25519. So I've aligned the size to 255.